src/cartsWithsSongs/services/carts_with_songs_service.ts

```diff
@@ -39,17 +40,25 @@ export class CartsWithSongsService {
     return this._cartRepository.delete({ id });
   }
 
-  public async getSongsFromCart(id) {
+  public async getSongsFromCart(id, isList) {
     const carts = await this._cartRepository.find({
       relations: ['owner', 'listOfSongs'],
     });
 
     const cart = await carts.find((item) => item?.owner?.id === id);
 
-    return this._songRepository.findByIds(
+    const songs = await this._songRepository.findByIds(
       [...cart.listOfSongs.map((song) => song.id)],
       { relations: ['author', 'genre'] }
     );
+
+    if (isList) {
+      songs.map(
+        async (song) => (song.image = await signedUrlMiddleware(song.image))
```

Comment:
You can map it to signedUrlMiddleware(song.image) functions, without 'await' and then resolve all promises in Promise.all() 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all